baremetal/metal3machine_manager_test.go

```diff
@@ -3626,6 +3627,12 @@ var _ = Describe("Metal3Machine manager", func() {
 			} else {
 				Expect(tc.M3Machine.Status.RenderedData).To(BeNil())
 			}
+			metal3DataReadyCondition := filterCondition(tc.M3Machine.Status.Conditions, infrav1.Metal3DataReadyCondition)
+			if tc.ExpectMetal3DataReadyCondition {
+				Expect(metal3DataReadyCondition).To(HaveLen(1))
+			} else {
+				Expect(metal3DataReadyCondition).To(HaveLen(0))
+			}
```

Comment:
instead of checking if it exists or no let's check if it is true or false

Reply:
fixed it kindly check.